### PR TITLE
[WIP] New method: xprod

### PR DESCRIPTION
**Pull request by Kelin2025**
- [x] Make performance test
- [ ] Maybe better solution to use `for`'s

### CODE REVIEW

File: lib/xprod/index.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-      })
-    )
-  }, [])
+  const res = []
+  for (var ia in a) {
+    for (var ib in b) {
+      res.push([a[ia], b[ib]])
+    }
+  }
+  return res
```
Comment (Beraliv) [inline lib/xprod/index.js]: Check the performance, please

**Comment by BabkinAleksandr** [comment]
I've added performance test. It's not good:

aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils__zsh_

**Comment by BabkinAleksandr** [comment]
I've updated performance test using performance helpers.

Also, I've speeded it up. Here is the tests:

| method | time |
| --- | --- |
| for-for | xprod - for-for |
| while-while | xprod - while-while |
| while-while improved* | xprods - while-while ext |

_* defining variables outside of loop, writing lengths to variables_

File: lib/xprod/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function xprod(a, b) {
+  const res = []
+  var ia = 0
+  var ib
+  var alen = a.length
+  var blen = b.length
+  while(ia < alen) {
```
Comment (Beraliv) [inline lib/xprod/index.js]: fix indents, please
below as well

**Comment by BabkinAleksandr** [comment]
Fixed syntax errors

**Comment by Beraliv** [comment]
Hooray, finally
@Kelin2025 and @BabkinAleksandr

**Pull request merged**